### DISCUSSION TITLE
transaction_key: deprecated and no transcription

### DISCUSSION THREAD

**Discussion by mcantillon21** [deepgram/community#288] [General help]
Hi All, 

New to deepgram and wondering why this issue persists: 

python deepgram_send.py

```
from flask import Flask, jsonify, request
from deepgram import Deepgram
import asyncio
import aiohttp

app = Flask(__name__)

@app.route('/', methods=['POST'])
def real_time_transcribe():
    loop = asyncio.new_event_loop()
    asyncio.set_event_loop(loop)
    
    if request.method == 'POST':
        with request.stream as stream:
            try:
                result = loop.run_until_complete(main(stream))
                response = jsonify(result)
                response.status_code = 200

            except Exception as e:
                response = jsonify({'error': str(e)})
                response.status_code = 500

    return response

async def main(stream):
    transcripts_list = []
    deepgram = Deepgram(DEEPGRAM_API_KEY)

    try:
        deepgramLive = await deepgram.transcription.live({ 'punctuate': True })
    except Exception as e:
        print(f'Could not open socket: {e}')
        return

    deepgramLive.registerHandler(deepgramLive.event.CLOSE, lambda c: print(f'Connection closed with code {c}.'))
    
    def handle_transcript_received(transcript):
        print(transcript)
        # print(transcript['channel']['alternatives'][0]['transcript'])
        transcripts_list.append(transcript)

    # Listen for any transcripts received from Deepgram and write them to the console
    deepgramLive.registerHandler(deepgramLive.event.TRANSCRIPT_RECEIVED, lambda transcript: handle_transcript_received(transcript))

    # Send streaming audio from the request stream to Deepgram
    chunk_size = 256
    
    while True:
        chunk = stream.read(chunk_size)
        if not chunk:
            break
        deepgramLive.send(chunk)

    # Indicate that we've finished sending data by sending the customary zero-byte message to the Deepgram streaming endpoint, and wait until we get back the final summary metadata object
    await deepgramLive.finish()
    
    return transcripts_list
  
if __name__ == "__main__":
    app.run(debug=True, port=5001)

```

and then here is my test, a synchronous stream of my mp3 file. 

```
import os
from pydub import AudioSegment
import requests


def stream_audio_to_server(audio_file, server_url):
    # Load audio file
    audio = AudioSegment.from_mp3(audio_file)
    
    # Convert to raw audio data
    raw_audio_data = audio.raw_data

    chunk_size = 1024

    for i in range(0, len(raw_audio_data), chunk_size):
        chunk = raw_audio_data[i:i + chunk_size]
        
        # Send chunk to server
        response = requests.post(server_url, data=chunk)

        # handle response here (if needed)
        print(response.text)


stream_audio_to_server('test.mp3', 'http://localhost:5001')

```

However I get the following Issues!

SERVER:
{'type': 'Metadata', 'transaction_key': 'deprecated', 'request_id': '77c9c92f-e742-41c9-b036-58478e20b9a0', 'sha256': '31de37f71992c499dd7f5500bcd8d3f41759698ba801696aea60954b1a1dfdd4', 'created': '2023-07-22T05:21:49.433Z', 'duration': 0.0, 'channels': 0}
Connection closed with code 1000.

TEST:
  {
    "channels": 0,
    "created": "2023-07-22T05:29:31.680Z",
    "duration": 0.0,
    "request_id": "acc6406d-8b1a-4ebe-8250-739c2431b730",
    "sha256": "cf831accf862108389ac79dffd0eb12e7e366c73e090cca0c22a7bdeef344c9c",
    "transaction_key": "deprecated",
    "type": "Metadata"
  }
]

Any idea what might be going on?

Fixed! Was a sample rate issue.

**Comment by jjmaldonis** [comment]
Hey @mcantillon21, you can ignore the `transcation_key: deprecated` field in the response. Check out this post for more info: https://github.com/orgs/deepgram/discussions/158

Typically an empty transcription means one of three things: 1) the audio is in a different language than the request being made to Deepgram (Deepgram assumes the language is English if it is not specified), 2) the audio is silent, or 3) the `encoding` and `sample_rate` parameters need to be set in the live transcription parameters (i.e. right next to `punctuate: True` in your code).

Here is a post about how to find the encoding of your mp3 file (the `ffprobe` command will also tell you the sample rate): https://stackoverflow.com/questions/5618363/is-there-a-way-to-use-ffmpeg-to-determine-the-encoding-of-a-file-before-transcod

Here is a link to our docs with valid values for the `encoding` parameter: https://developers.deepgram.com/docs/encoding

Can try setting the sample rate and encoding to see if that solves the issue?
